### PR TITLE
added lwip as a submodule

### PR DESCRIPTION
**Pull request by tridge**
this replaces #71 which adds lwip as a file import

### CODE REVIEW

**Comment by magicrub** [comment]
I've rebased this

File: os/various/lwip_bindings/arch/sys_arch.c [hunk 1]
```diff
@@ -261,13 +276,5 @@ void sys_arch_unprotect(sys_prot_t pval) {
...
-#if OSAL_ST_FREQUENCY == 1000
-  return (u32_t)chVTGetSystemTimeX();
-#elif (OSAL_ST_FREQUENCY / 1000) >= 1 && (OSAL_ST_FREQUENCY % 1000) == 0
-  return ((u32_t)chVTGetSystemTimeX() - 1) / (OSAL_ST_FREQUENCY / 1000) + 1;
-#elif (1000 / OSAL_ST_FREQUENCY) >= 1 && (1000 % OSAL_ST_FREQUENCY) == 0
-  return ((u32_t)chVTGetSystemTimeX() - 1) * (1000 / OSAL_ST_FREQUENCY) + 1;
-#else
-  return (u32_t)(((u64_t)(chVTGetSystemTimeX() - 1) * 1000) / OSAL_ST_FREQUENCY) + 1;
-#endif
+  return hrt_millis32();
```
Comment (tridge) [inline os/various/lwip_bindings/arch/sys_arch.c:279]: this needs ifdef _ARDUPILOT_

File: os/various/lwip_bindings/arch/sys_arch.c [hunk 2]
```diff
@@ -62,7 +62,7 @@
 #include "arch/cc.h"
 #include "arch/sys_arch.h"
 #include "lwipopts.h"
-
+#include "hrt.h"
```
Comment (tridge) [inline os/various/lwip_bindings/arch/sys_arch.c:65]: ifdef _ARDUPILOT_

File: os/various/lwip_bindings/arch/sys_arch.c (hunk 1 above)
Comment (magicrub) [reply os/various/lwip_bindings/arch/sys_arch.c:284]: fixed

File: os/various/lwip_bindings/arch/sys_arch.c (hunk 2 above)
Comment (magicrub) [reply os/various/lwip_bindings/arch/sys_arch.c:67]: fixed

**Review by magicrub — APPROVED** [review]
LGTM!!!!

**Pull request merged**